sqlitecluster/SQLite.cpp

```diff
@@ -526,3 +530,80 @@ size_t SQLite::getLastWriteChangeCount() {
     int count = sqlite3_changes(_db);
     return count > 0 ? (size_t)count : 0;
 }
+
+int SQLite::_sqliteAuthorizerCallback(void* pUserData, int actionCode, const char* detail1, const char* detail2,
+                                      const char* detail3, const char* detail4)
+{
+    SQLite* db = static_cast<SQLite*>(pUserData);
+    return db->_authorize(actionCode, detail1, detail2);
+}
+
+int SQLite::_authorize(int actionCode, const char* table, const char* column) {
+    // If the blacklist isn't set, we always return OK.
+    if (!blacklist) {
+        return SQLITE_OK;
+    }
+
+    switch (actionCode) {
+        // The following are *always* disallowed in blacklist mode.
+        case SQLITE_CREATE_INDEX:
+        case SQLITE_CREATE_TABLE:
+        case SQLITE_CREATE_TEMP_INDEX:
+        case SQLITE_CREATE_TEMP_TABLE:
+        case SQLITE_CREATE_TEMP_TRIGGER:
+        case SQLITE_CREATE_TEMP_VIEW:
+        case SQLITE_CREATE_TRIGGER:
+        case SQLITE_CREATE_VIEW:
+        case SQLITE_DELETE:
+        case SQLITE_DROP_INDEX:
+        case SQLITE_DROP_TABLE:
+        case SQLITE_DROP_TEMP_INDEX:
+        case SQLITE_DROP_TEMP_TABLE:
+        case SQLITE_DROP_TEMP_TRIGGER:
+        case SQLITE_DROP_TEMP_VIEW:
+        case SQLITE_DROP_TRIGGER:
+        case SQLITE_DROP_VIEW:
+        case SQLITE_INSERT:
+        case SQLITE_PRAGMA:
+        case SQLITE_TRANSACTION:
+        case SQLITE_UPDATE:
+        case SQLITE_ATTACH:
+        case SQLITE_DETACH:
+        case SQLITE_ALTER_TABLE:
+        case SQLITE_REINDEX:
+        case SQLITE_CREATE_VTABLE:
+        case SQLITE_DROP_VTABLE:
+        case SQLITE_SAVEPOINT:
+        case SQLITE_COPY:
+        case SQLITE_RECURSIVE:
+            return SQLITE_DENY;
+            break;
+
+        // The following are *always* allowed in blacklist mode.
+        case SQLITE_SELECT:
+        case SQLITE_ANALYZE:
+        case SQLITE_FUNCTION:
+            return SQLITE_OK;
+            break;
+
+        // The following is the only special case where the blacklist actually applies.
+        case SQLITE_READ:
```

Comment:
What's the difference between SQLITE_SELECT and SQLITE_READ? I couldn't find it here https://www.sqlite.org/c3ref/c_alter_table.html or in the code base